### PR TITLE
mds/mgr: fix handling of filesystem removal

### PR DESCRIPTION
**Pull request by jcsp**
The principal bug here was the FSMap constructor -- the dashboard fixes just make things a bit nicer.

Fixes: http://tracker.ceph.com/issues/21599

### CODE REVIEW

**Comment by batrick** [comment]
John, I'll leave this to you to QA?

**Comment by jcsp** [comment]
retest this please

File: src/pybind/mgr/dashboard/module.py [hunk 1]
```diff
@@ -255,6 +255,10 @@ def fs_status(self, fs_id):
                 filesystem = fs
                 break
 
+        if filesystem is None:
+            raise cherrypy.HTTPError(404,
```
Comment (nrdmn) [inline src/pybind/mgr/dashboard/module.py]: I'd raise http exceptions as late as possible, i.e. this one in the filesystem function.

**Comment by batrick** [comment]
@jcsp just checking if you're wanting me to QA this?

**Comment by jcsp** [comment]
@batrick ran this here: http://pulpito.ceph.com/jspray-2017-10-08_22:37:13-fs:basic_functional-wip-jcsp-testing-20171006-testing-basic-smithi/

It had a crash that looked unrelated to this PR: if they're familiar to you then this should be good to merge.
```
ceph version 13.0.0-1710-g02b1940 (02b1940444f922cce3afcf2cb2070619d85a5089) mimic (dev)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x110) [0x5615bfebf2c0]
 2: (MDCache::check_memory_usage()+0x7b9) [0x5615bfc59ab9]
 3: (MDSRankDispatcher::tick()+0xf7) [0x5615bfbb1e57]
 4: (FunctionContext::finish(int)+0x2a) [0x5615bfba15aa]
 5: (Context::complete(int)+0x9) [0x5615bfb9eaa9]
 6: (SafeTimer::timer_thread()+0x104) [0x5615bfebbb44]
 7: (SafeTimerThread::entry()+0xd) [0x5615bfebd58d]
 8: (()+0x7e25) [0x7fbbf00ace25]
 9: (clone()+0x6d) [0x7fbbef18f34d]
1 jobs: 1714327

```

**Comment by batrick** [comment]
... that looks new. I'll take a look.

**Comment by batrick** [comment]
@jcsp ya no way this is related. +1 to merge from me.

**Pull request merged**